quipucords/api/common/entities.py

```diff
@@ -125,12 +127,52 @@ def ip_addresses(self):
         with suppress(TypeError):
             return json.loads(self._fingerprints.ip_addresses)
 
+    @property
+    def ipv4_addresses(self):
+        """Retrieve only ipv4 addresses."""
+        ip_addresses = self.ip_addresses
+        ipv4_addresses = []
+        if ip_addresses:
+            for ip in ip_addresses:  # pylint: disable=invalid-name
+                try:
+                    if ipaddress.ip_address(ip).version == 4:
+                        ipv4_addresses.append(ip)
+                except ValueError:
+                    continue
+        return ipv4_addresses
+
+    @property
+    def ipv6_addresses(self):
+        """Retrieve only ipv6 addresses."""
+        # "Currently, we do not collect this information, but since it is
+        # mandatory for Yuptoo, we have agreed to send a default value."
+        # https://github.com/RedHatInsights/yuptoo/blob/265e9033ab66dce26d2a7bb5a909d7ec0b38a7da/yuptoo/modifiers/transform_network_interfaces.py#L8  # noqa: E501
+        return []
+
     @property
     def mac_addresses(self):
         """Retrieve mac_addresses."""
         with suppress(TypeError):
             return json.loads(self._fingerprints.mac_addresses)
 
+    @property
+    def mac_address(self):
+        """Retrieve and validate mac_address."""
+        mac_addresses = self.mac_addresses
+        mac_regex = r"^([0-9A-Fa-f]{2}[:-]){5}([0-9A-Fa-f]{2})$"
+        if mac_addresses:
+            if re.match(mac_regex, mac_addresses[0]):
+                return mac_addresses[0]
```

Comment:
what happens if the first mac address is invalid but the second is fine?
how about we iterate `mac_addresses` until we find one that works?

```suggestion
        for mac_addr in self.mac_addresses:
            if re.match(mac_regex, mac_addr):
                return mac_addr
```

Reply:
BTW, I don't love the fact we are randomly picking one from (possibly) many mac addresses, but I understand this is what insights requires.

We could also decide only send if there's only ONE valid mac_address among many, send if there's only ONE mac address and it is valid, not send et al, etc... Did you notice the pattern here? I'm talking about business rules.

Could you please check with @justinkreft whether this behavior is acceptable?

Reply:
I will bring those questions to Justin in the slack channel and update the pr either by removing the support for mac_address or adjusting the way it is validated.

Reply:
According to the conversation with Justin, I will remove mac_address from this pr.

Reply:
> BTW, I don't love the fact we are randomly picking one from (possibly) many mac addresses, but I understand this is what insights requires.
> 
> We could also decide only send if there's only ONE valid mac_address among many, send if there's only ONE mac address and it is valid, not send et al, etc... Did you notice the pattern here? I'm talking about business rules.
> 
> Could you please check with @justinkreft whether this behavior is acceptable?

we got an ok from justin and kevin regarding your concern. We are good to go.